sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -137,9 +139,9 @@ final class Decimal extends Ordered[Decimal] with Serializable {
 
   def toBigDecimal: BigDecimal = {
     if (decimalVal.ne(null)) {
-      decimalVal
+      decimalVal(new MathContext(Decimal.MAX_PRECISION, RoundingMode.HALF_EVEN))
     } else {
-      BigDecimal(longVal, _scale)
+      BigDecimal(longVal, _scale)(new MathContext(Decimal.MAX_PRECISION, RoundingMode.HALF_EVEN))
```

Comment:
`_precesion` is the current precision of this decimal. But here we want to assign a maximum precision that affects in later decimal operation like multiplication.